app/models/role_loader.rb

```diff
@@ -0,0 +1,45 @@
+class RoleLoader
+
+  def available_roles_for_application(name)
```

Comment:
The way the RoleLoader works internally needs a refactor agreed.
But its public interface is ok - instantiating a few Role models is no overhead (just PORO's) and provides a consistent API to RoleLoader

As long as the RoleLoader behaves correctly then there should not be leaking of roles as the application name must be declared (ie cannot get roles for all applications in 1 list)